src/dune_rules/foreign.ml

```diff
@@ -228,8 +261,11 @@ module Source = struct
 
   let path t = t.path
 
-  let object_name t =
-    t.path |> Path.Build.split_extension |> fst |> Path.Build.basename
+  let object_name ?(with_mode_suffix = true) t =
```

Comment:
Can you remove this optional argument? Thanks.

Reply:
7137b680337675b467a5f432bc99d497ba20b18b